python/GafferUI/ContextQueryUI.py

```diff
@@ -0,0 +1,408 @@
+
+		"queries.*" : [
+
+			"description",
+			"""
+			The default value to output if the variable does not exist.
+			""",
+
+			"spreadsheet:columnName", functools.partial( __getLabel, parentPlug = "queries", sanitise = True ),
+
+		],
+
+		"queries.*.name" : [
+
+			"description",
+			"""
+			The name of the variable to query.
+			""",
+
+		],
+
+		"queries.*.value" : [
+
+			"description",
+			"""
+			The value to output if the variable does not exist.
+			""",
+
+		],
+
+		"out" : [
+
+			"description",
+			"""
+			The parent plug of the query outputs. The order of outputs corresponds
+			to the order of children of `queries`.
+			""",
+
+			"plugValueWidget:type", "GafferUI.LayoutPlugValueWidget",
+
+			"layout:section", "Settings.Outputs",
+
+			"nodule:type", "GafferUI::CompoundNodule",
+			"noduleLayout:spacing", 0.4,
+
+		],
+
+		"out.*" : [
+
+			"description",
+			"""
+			The result of the query.
+			""",
+
+			"label", functools.partial( __getLabel, parentPlug = ""),
+
+			"plugValueWidget:type", "GafferUI.ContextQueryUI._OutputWidget",
+
+			"nodule:type", "GafferUI::CompoundNodule",
+
+		],
+
+		"out.*.exists" : [
+
+			"description",
+			"""
+			Outputs true if the variable exists in the context, and is a compatible type.
+			""",
+
+			"noduleLayout:label", functools.partial( __getLabel, parentPlug = "exists" ),
+
+		],
+
+		"out.*.value" : [
+
+			"description",
+			"""
+			Outputs the value of the specified variable, or the default value
+			if the variable does not exist ( or is incompatible ).
+			""",
+
+		],
+
+		"out.*.value..." : [
+
+			"noduleLayout:label", functools.partial( __getLabel, parentPlug = "values" ),
+
+		],
+
+	}
+)
+
+##########################################################################
+# _ContextQueryFooter
+##########################################################################
+
+class _ContextQueryFooter( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug ) :
+
+		row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+
+		GafferUI.PlugValueWidget.__init__( self, row, plug )
+
+		with row :
+
+			GafferUI.Spacer( imath.V2i( GafferUI.PlugWidget.labelWidth(), 1 ) )
+
+			GafferUI.MenuButton(
+				image = "plus.png",
+				hasFrame = False,
+				menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ) )
+			)
+
+			GafferUI.Spacer( imath.V2i( 1 ), imath.V2i( 999999, 1 ), parenting = { "expand": True } )
+
+		plug.node().plugSetSignal().connect(
+			Gaffer.WeakMethod( self.__updateQueryMetadata ),
+			scoped = False
+		)
+
+	def _updateFromPlug( self ) :
+
+		self.setEnabled( self._editable() )
+
+	def __menuDefinition( self ) :
+
+		result = IECore.MenuDefinition()
+
+		for item in [
+			Gaffer.BoolPlug,
+			Gaffer.FloatPlug,
+			Gaffer.IntPlug,
+			"NumericDivider",
+			Gaffer.StringPlug,
+			"StringDivider",
+			Gaffer.V2iPlug,
+			Gaffer.V3iPlug,
+			Gaffer.V2fPlug,
+			Gaffer.V3fPlug,
+			"VectorDivider",
+			Gaffer.Color3fPlug,
+			Gaffer.Color4fPlug,
```

Comment:
Let's extend this list to include all the types that the ContextVariablesUI allows - arrays etc. At some point we need to unify all this "menu for making plugs" code, but don't worry about that now. We can just manually match the list from ContextVariablesUI.